zebra-consensus/src/error.rs

```diff
@@ -33,7 +34,7 @@ pub enum SubsidyError {
     SumOverflow,
 }
 
-#[derive(Error, Clone, Debug, PartialEq, Eq)]
+#[derive(Error, Debug, PartialEq, Eq)]
```

Comment:
This was very helpful, rust-analyzer seemed to miss that `TransactionError` really needs to impl `Clone` because it's used for the rejection list in the mempool storage, but there's an `io::Error` variant in `HistoryTreeError` that was getting in the way of deriving Clone for `ValidateContextError`. 

We can just put the ValidateContextError in an Arc. Thank you.

Reply:
I would encourage you to put the `io::Error` in an `Arc`, so that everything that contains a `HistoryTreeError` can be `Clone`.